File: models/BrowserCategoryItem.go
package models

/* Представляет краткую информацию об упражнении,
	программе тренировок или диете, отображаемая в составе
	какой-либо категории (карточка)
*/
type BrowserCategoryItem struct {
	ID 		int					`db:"id"`
	Difficulty int				`db:"Difficulty"`
	Name 	string				`db:"Name"`
	Description string			`db:"Description"`
	ShortDescription string		`db:"ShortDescription"`
	Image 	string				`db:"Image"`
	Rating 	float32				`db:"Rating"`
	Voted   int					`db:"Voted"`
	Price 	int					`db:"Price"`

	/* Не используемые поля, необходимые для корректного получения результата
		данные поля присутствуют только в некоторых объктах и могут отсутствовать
		у других. */
	MuscleGroup int 			`db:"MuscleGroup" json:"-"`
	IsBase	bool				`db:"IsBase" json:"-"`
}

File: src/main/kotlin/com/breezesoftware/stayfit/models/BrowserCategoryItem.kt
package com.breezesoftware.stayfit.models

import java.security.KeyStore

/**
 * Содержит общие поля между всеми основными объектами,
 * такими как Программы тренировок, упражнения, данные
 * объекты предназначены для отображения в броузере программ
 * и упражнений.
 */
data class BrowserCategoryItem(
        var name : String,
        var description : String,
        var shortDescription : String,
        var image : String,
        var rating : Float,
        var voted : Long,
        var price : Int = 0
) {
    companion object {
        fun fromHashMap(map : Map<String, Any>) : BrowserCategoryItem {

            return BrowserCategoryItem(
                    name = map["Name"] as String,
                    description = map["Description"] as String,
                    shortDescription = map["ShortDescription"] as String,
                    image = map["Image"] as String,
                    rating = map["Rating"] as Float,
                    voted = map["Voted"] as Long
                    )
        }
    }
}